clap_complete/src/engine/complete.rs

```diff
@@ -165,6 +165,9 @@ fn complete_arg(
                 .find(|p| p.get_index() == Some(pos_index))
             {
                 completions.extend(complete_arg_value(arg.to_value(), positional, current_dir));
+                if matches!(positional.get_action(), clap::ArgAction::Append) {
```

Comment:
This works by happenstance for the current test cases but isn't the actual condition we should be checking for.  If `num_args(3..=10)`, then after `3` values, we should also report options.

We need to check if we have exceeded the minimum number of arguments for the positional.

Reply:
Actually, the current test cases demonstrate this bug.  `"pos_1 [TAB]"` hasn't met the minimum of `2` yet, and so should only report positionals.